src/test/java/test_extensions/ScreenshotRules.java

```diff
@@ -0,0 +1,55 @@
+package test_extensions;
+
+import driver.Config;
+import driver.DriverSingleton;
+import io.qameta.allure.Allure;
+import io.qameta.allure.Attachment;
+import io.qameta.allure.model.Parameter;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.TestWatcher;
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.OutputType;
+import org.openqa.selenium.TakesScreenshot;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+public class ScreenshotRules implements TestWatcher {
+
+    public WebDriver driver = DriverSingleton.getInstance().getDriver(Config.CHROME);
```

Comment:
please remove Driver from here, cause you can have issues with multithreading. get it on demand. and why do you get always Chrome driver... what about FF)?

Reply:
fixed